contracts/VoucherSets.sol

```diff
@@ -0,0 +1,299 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+pragma solidity 0.7.6;
+
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/token/ERC1155/IERC1155Receiver.sol";
+import "./interfaces/IVoucherKernel.sol";
+import "./interfaces/IVoucherSets.sol";
+import "./interfaces/ICashier.sol";
+
+import "hardhat/console.sol";
+
+//preparing for ERC-1066, ERC-1444, EIP-838
+
+/**
+ * @title Voucher sets implemented as ERC-1155
+ */
+contract VoucherSets is IVoucherSets, ERC1155, Ownable, ReentrancyGuard {
+    using SafeMath for uint256;
+    using Address for address;
+
+    address private voucherKernelAddress; //address of the VoucherKernel contract
+    address private cashierAddress; //address of the Cashier contract
+    
+
+    event LogVoucherKernelSet(address _newVoucherKernel, address _triggeredBy);
+    event LogCashierSet(address _newCashier, address _triggeredBy);
+
+    modifier onlyFromVoucherKernel() {
+        require(
+            voucherKernelAddress != address(0),
+            "UNSPECIFIED_VOUCHERKERNEL"
+        );
```

Comment:
Yes, but since it's not set in a constructor, this check is sort of needed until `voucherKernelAddress` is set.

Just adding it to a constructor leads to circular dependency, which I'm resolving in my other branch, so I can take care also of this case there. 
And I will remove this unnecessary requires at the same time.